framework/src/userobjects/TimedElementSubdomainModifier.C

```diff
@@ -0,0 +1,83 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// Base class to move elements to a specific subdomain at the given times.
+
+#include "TimedElementSubdomainModifier.h"
+
+registerMooseObject("MooseApp", TimedElementSubdomainModifier);
+
+InputParameters
+TimedElementSubdomainModifier::validParams()
+{
+  InputParameters params = ElementSubdomainModifier::validParams();
+  params.addParam<std::vector<Real>>("times", "The times of the subdomain modifications.");
+
+  return params;
+}
+
+TimedElementSubdomainModifier::TimedElementSubdomainModifier(const InputParameters & parameters)
+  : ElementSubdomainModifier(parameters)
+{
+}
+
+void
+TimedElementSubdomainModifier::initialize()
+{
+  // state variables
+  _last_t = -1;
+  _current_step = _t_step - 1;
+
+  // ask for all times (must NOT be sorted)
+  const auto _times = onGetTimes();
+  const auto n = _times.size();
+
+  // copy data to local storage
+  _times_and_indices.resize(n);
+  for (size_t i = 0; i < n; ++i)
+  {
+    _times_and_indices[i].time = _times[i];
+    _times_and_indices[i].index = i;
+  };
+
+  // sort the times
+  std::sort(_times_and_indices.begin(), _times_and_indices.end());
+}
+
+std::vector<Real>
+TimedElementSubdomainModifier::onGetTimes()
+{
+  std::vector<Real> _times;
+  return _times;
+}
+
+SubdomainID
+TimedElementSubdomainModifier::onComputeSubdomainID(Real /* t_from_exclusive */,
+                                                    Real /* t_to_inclusive */)
+{
+  const auto _subdomain_id = _current_elem->subdomain_id();
+  return _subdomain_id;
+}
```

Comment:
Also this one... You don't have to provide a default implementation. Just make the method pure virtual.

Reply:
Can you please describe in more detail what needs to be done here?

Reply:
See my comment above in the header file.